### PR TITLE
Feature/ld tensrst upd

### PR DESCRIPTION
**Pull request by sujayvkumar**
### Modifies the ensemble restart transformation capabilities in LDT

The ensemble restart generation has been modified to allow random sampling options when going from an restart file with certain number of ensembles to another restart file with a different number of ensemble members. 

### Testcase
Upscaling case:   /discover/nobackup/karsenau/LISF_PR/LDT_Ensrst_Upscale
 
Downscaling case:  /discover/nobackup/karsenau/LISF_PR/LDT_Ensrst_Downscale

### CODE REVIEW

**Comment by karsenau** [comment]
Yes, please find our test cases here:

Upscaling case:   /discover/nobackup/karsenau/LISF_PR/LDT_Ensrst_Upscale
 
Downscaling case:  /discover/nobackup/karsenau/LISF_PR/LDT_Ensrst_Downscale

File: ldt/core/LDT_ran2_gasdev.F90 [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+          k=idum/IQ1
+          idum=IA1*(idum-k*IQ1)-k*IR1
+          if (idum.lt.0) idum=idum+IM1
+          if (j.le.NTAB) iv(j)=idum
+       end do
+       iy=iv(1)
+    else
+       write (*,*) 'init_randseed(): initialize by calling with rseed(1)<0'
+       write (*,*) 'STOPPING.'
+       stop
```
Comment (bmcandr) [inline ldt/core/LDT_ran2_gasdev.F90:154]: Change:
* `write (*,*)` to `write (LDT_logunit, *)`
* `stop` to `LDT_endrun()`

@sujayvkumar @karsenau, I'm happy to make these changes if you approve. These fixes should also be made to `lis/core/LIS_ran2_gasdev.F90` (perhaps in a subsequent PR).

**Review by bmcandr — CHANGES_REQUESTED** [review]

**Comment by bmcandr** [comment]
I have finished reviewing and testing. I made two minor changes:

* Updated the copyright notice in `LDT_ran2_gasdev.F90`
* Added `Ensemble restart generation sampling strategy` to the LDT user guide. **Please review this entry.**

I have also requested a change to `LDT_ran2_gasdev.F90` indicated in the comment above.

Testing produced identical output with both Intel 19 and GNU9 compilers. Once the changes are reviewed and resolved this PR will be ready to merge.

File: ldt/core/LDT_ran2_gasdev.F90 (hunk 1 above)
Comment (karsenau) [reply ldt/core/LDT_ran2_gasdev.F90:154]: Hi @bmcandr, 

Yes, thank you -- that will be great. You can go ahead and make these changes.

Comment (karsenau) [reply ldt/core/LDT_ran2_gasdev.F90:154]: Just make sure to include the module 'use' call to 'LDT_logMod' for LDT_logunit

**Review by karsenau — APPROVED** [review]

File: ldt/core/LDT_ran2_gasdev.F90 (hunk 1 above)
Comment (bmcandr) [reply ldt/core/LDT_ran2_gasdev.F90]: Done!

**Comment by karsenau** [comment]
New updates made for the HyMAP2 Routing LDT-based ensemble restart support -- "random sampling"

Commit made to master branch (feature branch: feature/hymap_ensrst from karsenau) and to be pulled into the 557WW LIS7.4 support branch (@jvgeiger )

**Pull request merged**